firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Util/Array.txt

```diff
@@ -0,0 +1,25 @@
+SCENARIO("Create an array", "[array]")
+  GIVEN("A set of int values '\x0f\x36\x08\xff\x89\x01\x80\x02' ")
+  WHEN("An array of given int size is created properly")
+     THEN("Value at index 5 in array is as expected")
+     THEN("Size of an array is as expected")
+
+   GIVEN("set of values '\x01\x81\x01\x00' ")
+   WHEN("An array of 4UL size with   given values is created properly")
+     THEN("Value at index 5 in array is as expected")
+     THEN("Size of array is as expected")
+
+   GIVEN("A set of string '\ab\bc\cd' ")
+   WHEN("An array is created with given set of string")
+     THEN("Value at 0 index in array is as expected")
+     THEN("Size of an array is as expected")
+
+SCENARIO("Create an array and check data and size of array") 
+  GIVEN("An array ")
+   WHEN("Array is created with '\x01\x08\x0f\x0a'  ")
+     THEN("Value at 2 index in array is as expected")
+     THEN("Size of an array is as expected")
+
+  GIVEN("An array is given '\02\x03\x0a' ")
+  WHEN("Array is emptied")
+     THEN("The results are as expected")
```

Comment:
We should also test for the correctness of `make_array` when we pass in objects by value, as well as when we pass in const references to objects, as well as when we pass in mutable references to objects, because part of the complexity of the implementation of `make_array` is that it uses `std::forward` to handle the latter two cases, and because we use that functionality when we make the `initializables` array in `main.cpp`.